docs/DeveloperGuide.md

```diff
@@ -154,6 +154,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add/delete medical conditions from patients
+
+Users can add a medical condition to a particular patient by providing the following details:
+1. The patient's index number shown in the displayed patient list.
+2. The condition to be added.
+
+There are two ways a user can add a medical condition:
+1. Add multiple medical conditions at one go when the user first creates a patient.
+2. Add one condition at a time to an existing patient.
+
+#### Implementation
+
+A medical condition is represented by `Condition`, and multiple conditions are stored internally as a `ConditionList`.
+`ConditionList` facilitates the add/delete condition mechanism, and each patient only has one associated `ConditionList`.
+`ConditionList` mainly implements the following operations:
+* `ConditionList#add()`: adds a condition to the patient's list of conditions.
+* `ConditionList#delete()`: removes a condition from the patient's list of conditions.
+
+Figure 1 below summarises what happens when a user executes an add condition command on a specified patient:
+<figure>
+    <img src="images/AddConditionActivityDiagram.png" alt="add_condition_activity_diagram"/>
+    <figcaption>
+        <em>Figure 1: Activity diagram showing the flow of events when a user executes an add condition command</em>
```

Comment:
Figure number is temporary since it is unclear how many diagrams we will be using; will need to be updated for final version